carbon-design-system/carbon-addons-iot-react
# fix(hierarchy-list): show correct items on drag/drop w/ pagination

Closes #2911 

**Summary**

- The HierarchyList would reset to show the items from the first page after a drag-and-drop action was performed on another page.

**Change List (commits, features, bugs, etc)**

- take account the current page when setting which items to show in the useEffect

**Acceptance Test (how to verify the PR)**

- Go to the [HierarchyList with large number of items story](https://deploy-preview-2913--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-large-number-of-items)
- set the page size to sm
- go to page 2
- drag and drop an item
- list should stay on page 2 and still show items from page 2.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- no other side effects in other stories.



**Things to look for during review**
